test/test_buffer.cpp

```diff
@@ -26,9 +26,11 @@ namespace sparrow
     using buffer_test_type = buffer<uint8_t>;
     using view_test_type = buffer_view<uint8_t>;
 
+    using non_trivial_buffer_test_type = buffer<std::string>;
+
     namespace
     {
-        uint8_t* make_test_buffer(std::size_t size, uint8_t start_value = 0)
+        auto make_test_buffer(std::size_t size, uint8_t start_value = 0) -> uint8_t*
```

Comment:
I'm not completely sure but I suspect all the usage of `uint8_t` here (including previous code) can be replaced by `sparrow::byte_t` ?

Reply:
I'm ok to change that.
Are you ok @JohanMabille ?

Reply:
I finally changed that type to use a int32_t to avoid a sweet spot and to catch more errors in tests.

Reply:
I dont understand why specifically `int32_t` and not the type that represent bytes in the library? (also note that `start_value` needs to have that type too?)

Reply:
I choose int32_t because it's 4 bytes and not a single byte as previously and I expect to catch more issues with this case.
About `start_value`, it's the case, code is outdated in the preview.